File: src/main/java/net/softbell/bsh/domain/AuthStatusRule.java
package net.softbell.bsh.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 인증 상태 자료형
 */
@AllArgsConstructor
@Getter
public enum AuthStatusRule
{
	SUCCESS("SUCCESS", 0),
    FAIL("FAIL", 1),
    ERROR("ERROR", 2);

	private String value;
    private Integer code;
    
    public static AuthStatusRule ofLegacyCode(Integer legacyCode)
    {
    	for (AuthStatusRule authStatusRule : values())
            if (authStatusRule.getCode() == legacyCode)
                return authStatusRule;
    	
    	return null;
    }
}

File: src/main/kotlin/net/softbell/bsh/domain/AuthStatusRule.kt
package net.softbell.bsh.domain

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 인증 상태 자료형
 */
enum class AuthStatusRule {
    UNKNOWN { // 미확인
        override val value: String
            get() = "UNKNOWN"
        override val code: Int
            get() = -1
    },
    SUCCESS { // 정상 인증
        override val value: String
            get() = "SUCCESS"
        override val code: Int
            get() = 0
    },
    FAIL { // 인증 실패
        override val value: String
            get() = "FAIL"
        override val code: Int
            get() = 1
    };

    abstract val value: String
    abstract val code: Int

    companion object {
        fun ofLegacyCode(legacyCode: Int): AuthStatusRule {
            for (rule in values())
                if (rule.code == legacyCode)
                    return rule
            return UNKNOWN
        }
    }
}